addon/utils/cognito-user.js

```diff
@@ -11,24 +11,33 @@ export default EmberObject.extend({
 
   _callback(method, ...args) {
     return new Promise((resolve, reject) => {
-      this.get('user')[method](...args, (err, result) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(result);
-        }
-      });
+      try {
+        this.get('user')[method](...args, (err, result) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(result);
+          }
+        });
+      } catch (error) {
+        reject(error);
+      }
     }, `cognito-user#${method}`);
   },
 
   // Support for methods that user { onSuccess, onFailure } callback hashes
   _callbackObj(method, ...args) {
     return new Promise((resolve, reject) => {
-      this.get('user')[method](...args, {
-        onSuccess: resolve,
-        onFailure: reject
-      });
-    }, `cognito-user#${method}`);
+        try {
+          this.get('user')[method](...args, {
+            onSuccess: resolve,
+            onFailure: reject
+          });
+        } catch (error) {
+          reject(error);
+        }
+      }, `cognito-user#${method}`);
+
```

Comment:
Please check the indentation here --  it looks to be indented too much.

Reply:
adjusted